lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -1024,9 +1026,35 @@ namespace Js
         return RecyclerNewFinalized(recycler, ProjectionArrayBuffer, buffer, length, type);
     }
 
-    void ProjectionArrayBuffer::Dispose(bool isShutdown)
+    void ProjectionArrayBuffer::Finalize(bool isShutdown)
     {
+        if (isExternal)
+        {
+            return;
+        }
+
         CoTaskMemFree(buffer);
+
+        // In debugger scenario, ScriptAuthor can create scriptContext and delete scriptContext
```

Comment:
ScriptAuthor doesn't exist any more.
And it's always fine to get it from the library, especially in the Finalizer, as nothing is swept yet.

So you probably don't need this comment.

Reply:
I thought ScripAuthor is the user :-), did not know it was a thing. 
I just copied this piece from another finalizer.

I will modify comments in both places